archunit/src/main/java/com/tngtech/archunit/library/GeneralCodingRules.java

```diff
@@ -151,4 +154,28 @@ private GeneralCodingRules() {
     @PublicAPI(usage = ACCESS)
     public static final ArchRule NO_CLASSES_SHOULD_USE_JODATIME =
             noClasses().should(USE_JODATIME).because("modern Java projects use the [java.time] API instead");
+
+    /**
+     * For information about checking this condition, refer to {@link GeneralCodingRules}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchCondition<JavaField> USE_FIELD_INJECTION = useFieldInjection().as("use field injection");
+
+    private static ArchCondition<JavaField> useFieldInjection() {
+        ArchCondition<JavaField> annotatedWithAutowired = beAnnotatedWith("org.springframework.beans.factory.annotation.Autowired");
+        ArchCondition<JavaField> annotatedWithValue = beAnnotatedWith("org.springframework.beans.factory.annotation.Value");
+        ArchCondition<JavaField> annotatedWithInject = beAnnotatedWith("javax.inject.Inject");
+        ArchCondition<JavaField> annotatedWithResource = beAnnotatedWith("javax.annotation.Resource");
+        return annotatedWithAutowired.or(annotatedWithValue).or(annotatedWithInject).or(annotatedWithResource);
+    }
+
+    /**
+     * Field injection is seen as an anti-pattern.
+     * It is a good practice to use constructor injection for mandatory dependencies and setter injection for optional dependencies.
+     * <br>
+     * For information about checking this rule, refer to {@link GeneralCodingRules}.
+     */
+    @PublicAPI(usage = ACCESS)
+    public static final ArchRule NO_CLASSES_SHOULD_USE_FIELD_INJECTION =
+            noFields().should(USE_FIELD_INJECTION).because("field injection is considered harmful");
```

Comment:
Should we add more reason here? Maybe some link with more information? Because this way the "because" sounds more like a dogma "you should not do foo, because it's bad" :wink:

Reply:
We could also add the hints from the JavaDoc (_"use constructor injection for mandatory dependencies and setter injection for optional dependencies"_) to the `.because` to give the specific advice when the rule fails.